sdks/python/apache_beam/io/avroio.py

```diff
@@ -66,19 +68,22 @@ def __init__(self, label=None, file_pattern=None, min_bundle_size=0,
     Args:
       label: label of the PTransform.
       file_pattern: the set of files to be read.
+      coder: a coder to use to decode values (iff the record type is BYTES).
```

Comment:
The problem is that the Schema and the Coder are not quite the same thing. A Coder can be user-defined, and can encode any type of class (see the coder test, which writes a type not possible to store in Avro) but a Schema only handles primitive types but allows useful metadata to be stored in the file header itself (and hence is not required for reading). Limiting users to an AvroCoder would be unnecessarily restrictive, but not letting users provide a schema (where it works) would also be restrictive. 

Java is a bit different because things are strongly typed, and AvroCoders are constructed from a (simple enough) class object.